src/App.tsx

```diff
@@ -13,55 +13,28 @@ import {
   Text,
   VStack,
 } from "@chakra-ui/react";
-
+import usePublisher, { BroadcastOptions } from "./hooks/usePublisher";
 import IconCamera from "./components/Icons/Camera";
 import IconCameraOff from "./components/Icons/CameraOff";
+import { useLocation } from 'react-router-dom';
```

Comment:
We could replace the react-router-dom with web API, but please use it in the useEffect block of our usePublisher hook
```suggestion
const searchParams = new URLSearchParams(document.location.search);
const tokenValue = searchParams.get('token');
```